packages/webpack/mixins/optimizations/mixin.core.js

```diff
@@ -0,0 +1,102 @@
+const minimatch = require('minimatch');
+const { Mixin } = require('hops-bootstrap');
+
+function include(patterns) {
+  return (path) =>
+    // allow everything outside of node_modules
+    !path.includes('node_modules') ||
+    // allow all hops packages
+    /node_modules[/\\]hops/.test(path) ||
+    // check if any other rule matches
+    patterns.some((rule) => minimatch(path, rule));
+}
+
+function unique(value, index, self) {
+  return self.indexOf(value) === index;
+}
+
+function findResolved(a, b) {
+  try {
+    return a === require.resolve(b);
+  } catch {
+    return false;
+  }
+}
+
+class FastDevWebpackMixin extends Mixin {
+  configureBuild(webpackConfig, { allLoaderConfigs, jsLoaderConfig }, target) {
+    const { fastBuild, fastDev } = this.options;
+
+    if (!fastDev || !fastBuild) {
+      return;
+    }
+
+    const { NODE_ENV } = process.env;
+    const isDevelop = target === 'develop' || NODE_ENV !== 'production';
+    const optimizeNoCoreJs = fastDev && isDevelop;
+    const optimizeCoreJsEntry = !optimizeNoCoreJs && fastBuild;
+
+    if (optimizeNoCoreJs || optimizeCoreJsEntry) {
+      const {
+        experimental: { fastBuild: { babelIncludePatterns = [] } = {} } = {},
+      } = this.config;
+
+      jsLoaderConfig.include = include(babelIncludePatterns);
+      jsLoaderConfig.exclude = [];
+
+      webpackConfig.entry = []
+        .concat(webpackConfig.entry)
+        .filter((entry) => !/(core-js|regenerator-runtime)/.test(entry));
+
+      const [, presetEnvOptions] =
+        jsLoaderConfig.options.presets.find(([path]) => {
+          return findResolved(path, '@babel/preset-env');
+        }) || [];
+
+      if (presetEnvOptions) {
+        if (optimizeNoCoreJs) {
+          presetEnvOptions.useBuiltIns = false;
+          delete presetEnvOptions.corejs;
+        } else if (optimizeCoreJsEntry) {
+          presetEnvOptions.useBuiltIns = 'entry';
+
+          webpackConfig.entry = [
+            require.resolve('core-js/stable'),
+            require.resolve('regenerator-runtime/runtime'),
+          ]
+            .concat(webpackConfig.entry)
```

Comment:
Why do you use the spread-syntax in objects, but not in arrays? 😜

Reply:
Because `webpackConfig.entry` isn't always an array.